packages/smart-wallet/src/smartWallet.js

```diff
@@ -156,405 +147,428 @@ export const initState = (unique, shared) => {
+      updateBalance: M.call(PurseShape, AmountShape).optional('init').returns(),
+      publishCurrentState: M.call().returns(),
+      addBrand: M.call(
+        {
+          brand: BrandShape,
+          issuer: M.eref(IssuerShape),
+          petname: M.string(),
+          displayInfo: DisplayInfoShape,
+        },
+        M.eref(PurseShape),
+      ).returns(M.promise()),
+    }),
+    deposit: M.interface('depositFacetI', {
+      receive: M.callWhen(M.await(M.eref(PaymentShape))).returns(AmountShape),
+    }),
+    offers: M.interface('offers facet', {
+      executeOffer: M.call(shape.OfferSpec).returns(M.promise()),
+      getLastOfferId: M.call().returns(M.number()),
+    }),
+    self: M.interface('selfFacetI', {
+      handleBridgeAction: M.call(shape.StringCapData, M.boolean()).returns(
+        M.promise(),
+      ),
+      getDepositFacet: M.call().returns(M.remotable()),
+      getOffersFacet: M.call().returns(M.remotable()),
+      getCurrentSubscriber: M.call().returns(M.remotable()),
+      getUpdatesSubscriber: M.call().returns(M.remotable()),
+    }),
   };
-};
 
-const behaviorGuards = {
-  helper: M.interface('helperFacetI', {
-    updateBalance: M.call(PurseShape, AmountShape).optional('init').returns(),
-    publishCurrentState: M.call().returns(),
-    addBrand: M.call(
-      {
-        brand: BrandShape,
-        issuer: M.eref(IssuerShape),
-        petname: M.string(),
-        displayInfo: DisplayInfoShape,
-      },
-      M.eref(PurseShape),
-    ).returns(M.promise()),
-  }),
-  deposit: M.interface('depositFacetI', {
-    receive: M.callWhen(M.await(M.eref(PaymentShape))).returns(AmountShape),
-  }),
-  offers: M.interface('offers facet', {
-    executeOffer: M.call(shape.OfferSpec).returns(M.promise()),
-    getLastOfferId: M.call().returns(M.number()),
-  }),
-  self: M.interface('selfFacetI', {
-    handleBridgeAction: M.call(shape.StringCapData, M.boolean()).returns(
-      M.promise(),
-    ),
-    getDepositFacet: M.call().returns(M.remotable()),
-    getOffersFacet: M.call().returns(M.remotable()),
-    getCurrentSubscriber: M.call().returns(M.remotable()),
-    getUpdatesSubscriber: M.call().returns(M.remotable()),
-  }),
-};
-
-const SmartWalletKit = defineVirtualExoClassKit(
-  'SmartWallet',
-  behaviorGuards,
-  initState,
-  {
-    helper: {
-      /**
-       * @param {RemotePurse} purse
-       * @param {Amount<any>} balance
-       * @param {'init'} [init]
-       */
-      updateBalance(purse, balance, init) {
-        const { purseBalances, updatePublishKit } = this.state;
-        if (init) {
-          purseBalances.init(purse, balance);
-        } else {
-          purseBalances.set(purse, balance);
-        }
-        updatePublishKit.publisher.publish({
-          updated: 'balance',
-          currentAmount: balance,
-        });
-        const { helper } = this.facets;
-        helper.publishCurrentState();
-      },
+  return defineDurableExoClassKit(
```

Comment:
why not use `prepareExoClassKit`?  Shouldn't this be using `prepareKindHandle`, which that uses automatically?

If it's correct to not use prepare, then a comment here giving the reason is probably worthwhile.

Reply:
thanks I forgot about that one

Reply:
I see this comment resolved, and a request for a new review, but there's no new commit, AFAICT.